### PR TITLE
feat(buffer): add FindNext Motion through buffet/hotsauce

### PR DESCRIPTION
**Pull request by elkowar**
This PR adds the FindNext motion by using https://github.com/buffet/hotsauce as a regex crate

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -129,9 +130,9 @@ checksum = "dfb24e866b15a1af2a1b663f10c6b6b8f397a84aadb828f12e5b289ec23a3a3c"
 
 [[package]]
 name = "cc"
-version = "1.0.77"
```
Comment (buffet) [inline Cargo.lock:132]: Can you update in a separate commit?

File: crates/bazed-core/src/user_buffer_op.rs [hunk 2]
```diff
@@ -48,4 +48,6 @@ pub(crate) enum Motion {
     BottomOfViewport,
     NextWordBoundary(WordBoundaryType),
     PrevWordBoundary(WordBoundaryType),
+    FindNext(hotsauce::Regex),
```
Comment (buffet) [inline crates/bazed-core/src/user_buffer_op.rs:51]: These should take &Regex (&RegexCursor?) to allow stuff like caches.

Would also allow Copy again.

**Review by buffet — CHANGES_REQUESTED** [review]

**Comment by elkowar** [comment]
Both fixed!

**Review by buffet — APPROVED** [review]
LGTM!

**Pull request merged**